pkg/lib/bundle/supported_resources.go

```diff
@@ -33,6 +35,8 @@ var supportedResources = map[string]Namespaced{
 	ServiceMonitorKind:     true,
 	SecretKind:             true,
 	ConfigMapKind:          true,
+	DeploymentKind:         true,
+	APIServiceKind:         true,
```

Comment:
Confused here. Is this just so the validator doesn't fail when these files are included in the bundle? I thought the bundle would just contain the CSV as normal, if that's the case then why wouldn't we just ignore these files in the bundle itself?

And wouldn't it potentially contain lots of other arbitrary kube types that it seems like we wouldn't support?

Reply:
> Had a few comments.
> 
> Additionally, can this include a detailed doc with use cases and examples?
> 
> Also, where are we with integration between this tooling and the sdk?

Thanks for the comments.
The doc will be from another PR https://github.com/operator-framework/operator-registry/pull/287. I have made some local changes and will update once this PR is merged. Because so many changes have occurred in this PR, writing a doc every time there is a comment to change the behavior is a lot of redundant work. Thus, this is the approach I'm taking.

I don't really know the second question, but Eric has been reviewing this PR as well, so I think at least they know what is going on.

Reply:
Shouldn't all the supported kinds be listed here? This is an effort that came before this work. In this work, we have to extend the supported types here so that the kinds that are required for CSV are allowed to pass validation. 

What other kinds of file you expect? Since they are not included in CSV, and was not accepted by our previous validation definiation, I don't think we would allow them because we really don't support them.